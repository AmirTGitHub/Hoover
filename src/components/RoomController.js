import React from 'react';
import Input from './Input'

const RoomController = (props) => {

    return (
        <div>
            <div className='centre'>
                <Input title='Row' name='row' getRow={props.getRow} disabled={props.disabled} />
                <Input title='Column' name='column' getColumn={props.getColumn} disabled={props.disabled} />
            </div>
            <div className='centre' >
                {props.display ?
                    <Input title='Dirt position' name='dirt' getDirt={props.getDirt} placeholder={`please write a number between 0 and ${props.maxNumber}`} />
                    : null}
                <button className='btn' onClick={props.createRoom} > Create the room </button>

            </div>

        </div>
    )
}
export default RoomController