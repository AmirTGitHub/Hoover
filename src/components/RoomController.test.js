import RoomController from './RoomController'


it('render the three inputs with one button', () => {
    const wrapper = render(
        <RoomController />
    )
    expect(wrapper).toMatchSnapshot()
})



it('button works correctly', () => {
    const onClick = jest.fn()
    const wrapper = mount(<RoomController createRoom={onClick} />)
    const btn = wrapper.find('.btn').simulate('click')
    expect(onClick).toHaveBeenCalledTimes(1)


})

