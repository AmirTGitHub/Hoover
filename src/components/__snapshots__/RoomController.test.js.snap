// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the three inputs with one button 1`] = `
<div>
  <div
    class="centre"
  >
    <form
      style="padding:50px"
    >
      <h3>
        Row
      </h3>
      <label>
        
      </label>
      <input
        data-test="input"
        name="row"
        placeholder="number"
        type="number"
      >
        
      </input>
    </form>
    <form
      style="padding:50px"
    >
      <h3>
        Column
      </h3>
      <label>
        
      </label>
      <input
        data-test="input"
        name="column"
        placeholder="number"
        type="number"
      >
        
      </input>
    </form>
  </div>
  <div
    class="centre"
  >
    <button
      class="btn"
    >
       Create the room 
    </button>
  </div>
</div>
`;
